### PR TITLE
Add basic compatibility with marathon-lb

### PR DESCRIPTION
**Pull request by guilhem**
Add compatibility with labels: `HAPROXY_GROUP` and `HAPROXY_0_VHOST`.
* `HAPROXY_GROUP` become a new tag
* `HAPROXY_0_VHOST` become a rule `Host:`

https://github.com/mesosphere/marathon-lb

### CODE REVIEW

**Comment by guilhem** [comment]
I'm hopping this to be merged in 1.1 :angel:

**Comment by guilhem** [comment]
@emilevauge done :)
don't hesitate to tell me if this isn't enough

File: provider/marathon_test.go [hunk 1]
```diff
@@ -1152,37 +1201,53 @@ func TestMarathonGetEntryPoints(t *testing.T) {
...
 		},
 		{
 			application: marathon.Application{
 				Labels: &map[string]string{
 					"traefik.frontend.rule": "Host:foo.bar",
+					"HAPROXY_0_VHOST":       "notvalid",
+				},
+			},
+			marathonLBCompatibility: true,
+			expected:                "Host:foo.bar",
```
Comment (emilevauge) [inline provider/marathon_test.go]: Not sure to understand here :thinking:
I was expecting `expected:                "Host:notvalid"` instead here reading the code in `getFrontendRule`. Am I missing something?

Comment (guilhem) [inline provider/marathon_test.go]: if `traefik.frontend.rule` is present there is a `return` before trying `HAPROXY` label.

I prefer this behavior with `traefik` labels superseding other one.

Comment (emilevauge) [inline provider/marathon_test.go]: OK thanks :)

**Comment by emilevauge** [comment]
@guilhem needs rebase on v1.1 branch :)

**Comment by guilhem** [comment]
@emilevauge done :)

**Comment by guilhem** [comment]
I think you can squash with 1 button (to remove "merge" commit)

**Pull request merged**